File: firebase_test/backend/src/models/chat-model.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const chatSchema = new Schema({
  content: {
    type: String,
    required: true,
  },
  sequence: {
    type: Number,
    required: true,
  },
  users: [
    {
      type: Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
  ],
});

const Chat = mongoose.model('Chat', chatSchema);

module.exports = Chat;

File: firebase_test/app/lib/models/chat_model.dart
class Chat {
  final String id;
  final String content;
  final int sequence;

  Chat({required this.id, required this.content, required this.sequence});

  Chat.fromJson(Map<String, dynamic> data)
      : id = data['_id'],
        content = data['content'],
        sequence = data['sequence'];
}
